purchase_sale_inter_company/models/purchase_order.py

```diff
@@ -172,14 +181,25 @@ def _prepare_sale_order_line_data(self, purchase_line, dest_company, sale_order)
         return new_line._convert_to_write(new_line._cache)
 
     def button_cancel(self):
-        sale_orders = (
```

Comment:
Here, to be less invasive to the code and also not introduce a loop, you could do instead:

```
purchase_orders = self.filtered(lambda p: not getattr(p, "auto_sale_order_id", False))
(...)
.search([("auto_purchase_order_id", "in", purchase_orders.ids)])
(...)
purchase_orders.write({"partner_ref": False})
```